gitops/snapshot.go

```diff
@@ -320,6 +320,7 @@ func NewSnapshot(application *applicationapiv1alpha1.Application, snapshotCompon
 			Components:  *snapshotComponents,
 		},
 	}
+	go metrics.RegisterNewSnapshot()
```

Comment:
Won't this register a new Snapshot even when we realize that there is a matching snapshot that already exists like e.g. [here](https://github.com/redhat-appstudio/integration-service/blob/8075f54853a135480cf43610e93032eb66d34e62/controllers/pipeline/pipeline_adapter.go#L100)?

Reply:
We ever discussed it before. Currently, I don't have a good solution for it.